### ISSUE TITLE
[avy] Opening URLs via avy (SPC j U) is broken

### ISSUE THREAD

**Issue by nwolfe** [syl20bnr/spacemacs#4604]
On current `develop`, `SPC j U` yields `Symbol's function definition is void: spacemacs/avy-goto-url` and does nothing.  

Only after I invoke some other avy function does this function get defined. So if I call `SPC SPC spacemacs/avy-open-url`, subsequent `SPC j U` presses do find the function `spacemacs/avy-goto-url`....but that doesn't appear to do anything.

To reproduce:
1. Open Emacs
2. Put a couple URLs in scratch buffer, like `https://www.google.com`
3. Invoke `SPC j U` to see missing function message
4. Invoke `SPC SPC spacemacs/avy-open-url` to see correct behavior
5. Invoke `SPC j U` again to see what appears to be correct behavior, but hitting a link doesn't do anything

It seems like we should just bind `SPC j U` to `spacemacs/avy-open-url` -- not sure what the difference between these two functions is.

#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.8
- Spacemacs branch: develop (rev. 8adb066)
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
((auto-completion :variables auto-completion-enable-help-tooltip t auto-completion-enable-snippets-in-popup t)
 better-defaults
 (clojure :variables clojure-enable-fancify-symbols t)
 (colors :variables colors-enable-rainbow-identifiers t)
 deft emacs-lisp erc eyebrowse git github markdown
 (mu4e :variables mu4e-installation-path "/usr/local/share/emacs/site-lisp")
 org osx puppet ruby
 (shell :variables shell-default-shell 'eshell)
 spacemacs-layouts spotify
 (spell-checking :variables spell-checking-enable-by-default nil)
 version-control yaml)

**Comment by StreakyCobra** [comment]
At the beginning `SPC x o` was `avy-open-url` and this function was autoloaded with `:commands`. In a @justbur PR that grouped jumping functions to `SPC j`, he moved and replaced it by `avy-goto-url` that jump to the URL, but doesn't open it. This function just has to be added to `:commands` to avoid the loading bug, it's an easy fix.

I already pointed the missing `avy-open-url` to @justbur and he is open to change `SPC j U` to it.

From my point of view though, `SPC j U` can stay `avy-goto-url` as it's a *jumping* function, and we can restore `SPC x o` as `avy-open-url`, as it's a mode-agnostic way to open an url (it's just based on the texts inside the buffer) and as it doesn't jump to the URL.

In any way I'm open to any solution, as I already remapped it to my prefix: `SPC o u`. That fits really well for the mnemonic as **o**pen **u**rl :-)

:+1: Thanks for bringing the discussion, I forgot to do it.

**Comment by nwolfe** [comment]
> From my point of view though, SPC j U can stay avy-goto-url as it's a jumping function, and we can restore SPC x o as avy-open-url, as it's a mode-agnostic way to open an url (it's just based on the texts inside the buffer) and as it doesn't jump to the URL.

Ahh, I understand the distinction now.  This makes sense to me, to leave `SPC j U` as a cursor-movement command and restore `SPC x o` to actually open URLs.

**Comment by justbur** [comment]
For the record, I think it's fine to have `jU` be "jump to what the url points to", meaning it can open it. I can't remember now, but it's possible I just picked the wrong avy function when I was doing this.

**Comment by syl20bnr** [comment]
Removing `SPC x o` is a mistake on my side, I did not see that `SPC j U` was just to jump to a url which makes sense.

**Comment by StreakyCobra** [comment]
If a new contributor want to tackle this before us :smiley: https://github.com/syl20bnr/spacemacs/commit/fa50f4600a34172d77bf566e8c32221ec24935fe#diff-2c19a76d526eca441553bdf7c559d5abL402

**Comment by nwolfe** [comment]
I think I fixed it: #4621
